tools/js-sdk-release-tools/src/common/utils.ts

```diff
@@ -298,6 +299,31 @@ export async function resolveOptions(typeSpecDirectory: string): Promise<Exclude
     return options
 }
 
+export function updateApiVersionInTspConfig(typeSpecDirectory: string, apiVersion?: string) {
+    if (!apiVersion || apiVersion === '') {
+        logger.warn(`No apiVersion provided, skipping update to tspconfig.yaml.`);
+        return;
+    }
+    
+    const tspConfigPath = path.join(typeSpecDirectory, 'tspconfig.yaml');
+    if (!fs.existsSync(tspConfigPath)) {
+        throw new Error(`tspconfig.yaml not found at path: ${tspConfigPath}`);
+    }
+
+    const tspConfigContent = fs.readFileSync(tspConfigPath, 'utf8');
+    const tspConfig = yamlLoad(tspConfigContent);
+
+    if (!tspConfig || typeof tspConfig !== 'object') {
+        throw new Error('Invalid tspconfig.yaml format.');
+    }
+
+    tspConfig['options']['@azure-tools/typespec-ts']['api-version'] = apiVersion;
+
+    const updatedTspConfigContent = dump(tspConfig);
+    fs.writeFileSync(tspConfigPath, updatedTspConfigContent, 'utf8');
+    logger.info(`Updated API version to ${apiVersion} in tspconfig.yaml in pipeline only.`);
+}
```

Comment:
when tspconfig has `api-version`, and it is not the same to pipeline input: use warning logging that it will be override by pipeline input.